src/nodes/modify_hypertable_exec.c

```diff
@@ -2249,6 +2250,13 @@ ExecModifyTable(CustomScanState *cs_node, PlanState *pstate)
 	resultRelInfo = node->resultRelInfo + node->mt_lastResultIndex;
 	subplanstate = outerPlanState(node);
 
+	/*
+	 * Check for frozen chunk DML operation.
+	 * INSERTS are blocked in chunk tuple routing.
+	 */
+	if (operation != CMD_INSERT)
+		checkDMLOnFrozenChunk(resultRelInfo);
```

Comment:
does resultRelInfo always point to something meaningful at this point? Isnt it just the hypertable on first iteration?